pkg/repository/product/productRepositoryImpl.go

```diff
@@ -23,13 +25,16 @@ func (r *productRepositoryImpl) Create(ctx context.Context, product *entities.Pr
 	ctx, sp := tracer.Start(ctx, "productCreateRepository")
 	defer sp.End()
 
-	modelProduct := ToProductModel(product)
+	modelProduct := r.ToProductModel(product)
 	newProduct := new(model.Product)
 
 	if err := r.db.Connect().Create(modelProduct).Scan(newProduct).Error; err != nil {
 		return nil, errors.New(fmt.Sprintf("failed to create product"))
 	}
-	return newProduct.ToProductEntity(), nil
+
+	productEntity := newProduct.ToProductEntity()
+	r.SetProductSubAttributes(newProduct, sp)
+	return productEntity, nil
 }
 
 func (r *productRepositoryImpl) FindAll(ctx context.Context) (*[]entities.Product, error) {
```

Comment:
**ctx** is not used in this function

Reply:
it last next func in app,  not pass to func next  , i will fix ctx,sp > _,sp